src/comm/ModDevPoll.cc

```diff
@@ -48,19 +45,15 @@
 
 #define DEBUG_DEVPOLL 0
 
-// OPEN_MAX is defined in <climits>
-#define DEVPOLL_UPDATESIZE  OPEN_MAX
-#define DEVPOLL_QUERYSIZE   OPEN_MAX
+// Maximum number of fd's to buffer requests to change polling state
+#define DEVPOLL_UPDATESIZE  256
```

Comment:
My thought was simply that why should we allocate more memory than we might need, but as these two allocations each take only 64-bytes per fd, the default 4095 fd's will only add up to 256kb for each, and the apparently maximum of 32768 would only be 2mb for each, both of which were a ton of memory when I started programming, but are likely not so noticeable on a modern server running squid.  I can simplify down to Squid_MAXFD for both.

Reply:
> why should we allocate more memory than we might need

... because Squid might need all of it (and Squid will crash if we do not allocate enough memory for these arrays).

It would be possible to refactor Squid code so that it wastes less memory without crashing. However, that refactoring requires a lot of work, saving a few megabytes of RAM is not a sufficient incentive to perform that work IMO, and this PR is not about that kind of optimization.